lib/services/ride-comparison.ts

```
@@ -98,9 +85,69 @@ const WAYMO_SERVICE_AREAS = {
-/**
- * Check if coordinates are within Waymo's service area
- */
+function cleanupCache<T>(cache: Map<string, { value: T; expiresAt: number }>): void {
+  const now = Date.now()
+  const keysToDelete: string[] = []
+  cache.forEach((entry, key) => {
+    if (entry.expiresAt <= now) {
+      keysToDelete.push(key)
+    }
+  })
+  keysToDelete.forEach(key => cache.delete(key))
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)